CONTRIBUTING.md

```diff
@@ -0,0 +1,61 @@
+# Contribution Guidelines
+
+Thank you for your interest in contributing to Unity Robotics! To facilitate your 
+contributions, we've outlined a brief set of guidelines to ensure that your extensions 
+can be easily integrated.
+
+## Communication
+
+First, please read through our
+[code of conduct](CODE_OF_CONDUCT.md),
+as we expect all our contributors to follow it.
+
+Second, before starting on a project that you intend to contribute to any of our
+Unity Robotics packages or tutorials, we **strongly** recommend posting on the repository's
+[Issues page](https://github.com/Unity-Technologies/ROS-TCP-Endpoint/issues) and
+briefly outlining the changes you plan to make. This will enable us to provide
+some context that may be helpful for you. This could range from advice and
+feedback on how to optimally perform your changes or reasons for not doing it.
+
+Lastly, if you're looking for input on what to contribute, feel free to reach
+out to our team directly at [unity-robotics@unity3d.com](mailto:unity-robotics@unity3d.com) 
+and/or browse the GitHub issues.
+
+## Git Branches
+
+The `main` branch corresponds to the most recent stable version of the project. The `dev` branch
+contains changes that are staged to be merged into `main` as the team sees fit.
+
+When contributing to the project, please make sure that your Pull Request (PR)
+does the following:
+
+- Is up-to-date with and targets the `dev` branch
+- Contains a detailed description of the changes performed
+- Has corresponding changes to documentation, unit tests and sample environments (if
+  applicable)
+- Contains a summary of the tests performed to validate your changes
+- Links to issue numbers that the PR resolves (if any)
+
+<!-- ## Continuous Integration (CI)
+
+We run continuous integration on all PRs; all tests must be passing before the PR is merged. -->
+
+### Code style
+
+All Python code should follow the [PEP 8 style guidelines](https://pep8.org/).
+
+All C# code should follow the [Microsoft C# Coding Conventions](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions).
```

Comment:
Might want to point out here that we expect submissions to follow these conventions even if the code they are changing does not, and also they should be aware of the [Unity coding package](https://github.com/Unity-Technologies/com.unity.coding/blob/master/UnityCoding/Packages/com.unity.coding/README.md) since we have a couple special conventions that will supersede the standard C# ones.

Reply:
Will add the note about following conventions even if the code they're changing does not! Re: coding package, I think that package is private though, and I haven't been able to find a public-facing version of Unity standards.

Reply:
Hmm yeah, you're right... well, the [Unity Coding package](https://docs.unity3d.com/Packages/com.unity.coding@0.1/manual/index.html) itself is public;  that might be the best resource we can provide.